test/View/Helper/FormDateSelectTest.php

```diff
@@ -115,4 +117,28 @@ public function testGetElements(): void
 
         $this->assertCount(31, $elements[0]->getValueOptions());
     }
+
+    /**
+     * @group issue-160
+     * @throws ReflectionException
+     */
+    public function testParsePatternWithShortDateFormat(): void
+    {
+        $element  = new DateSelect('foo');
+        $dateType = IntlDateFormatter::SHORT;
+        $locale   = 'es_CL';
+
+        $this->helper->setLocale($locale);
+        $this->helper->setDateType($dateType);
+
+        $method = new ReflectionMethod($this->helper, 'parsePattern');
```

Comment:
Hmm, this looks even worse than before - we shouldn't test internal methods via reflection.

Also here, when reading this test out of context, it is unclear what you are verifying.

Here's my understanding:

 * when using `es_CL`
 * and setting a short date format
 * when I parse the pattern
 * then the pattern has day, month, year

So far, so good, but what's the effect on the public-facing behavior of this element helper?

Reply:
As for the commit message: `laminas#160: escape regex-meta-character in AbstractFormDateSelect `still doesn't say much to the reader :|

Reply:
I'm escaping two regex meta characters which are currently unescaped, like descriped in issue #160. I dont know how to test this without reflection as the functions which are causing the error are protected.

Reply:
This should be tested via public behavior. If there is no public behavior that exposes the bug then it is fine to fix without providing a test.

Reply:
> then it is fine to fix without providing a test.

No, it is not: it means we have dead/unreachable code then, and it should just be dropped or refactored instead.
Bugfixing without a valid test is just more maintenance burden for upstream maintainers.

Reply:
Checking this out (locally) to re-design the test, and try to trip the issue with `es_CL`.

Reply:
I've pushed faefb5bbe578a071b95f3a34a145940a28b2acee, which solves this discussion, and provides some context of what is going on (for posterity).

Meanwhile, thanks for the patience, @mimmi20!